compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -607,37 +607,34 @@ class Semantic {
       if classRef.memberClasses.nonEmpty || !warm.isFullyFilled then
         return PromoteError(msg, source, trace.toVector) :: Nil
 
-      val fields  = classRef.fields
-      val methods = classRef.membersBasedOnFlags(Flags.Method, Flags.Deferred | Flags.Accessor)
       val buffer  = new mutable.ArrayBuffer[Error]
 
-      fields.exists { denot =>
-        val f = denot.symbol
-        if !f.isOneOf(Flags.Deferred | Flags.Private | Flags.Protected) && f.hasSource then
-          val trace2 = trace.add(f.defTree)
-          val res = warm.select(f, source)
-          locally {
-            given Trace = trace2
-            buffer ++= res.ensureHot(msg, source).errors
+      warm.klass.baseClasses.exists { klass =>
+        klass.hasSource && klass.info.decls.exists { member =>
+          if member.isOneOf(Flags.Method | Flags.Lazy | Flags.Deferred) then {
```

Comment:
Remove `| Flags.Deferred`: it means the member is only declared but not defined.

I'd write:

```Scala
if !member.isType && !member.isConstructor && member.hasSource  && !member.is(Flags.Deferred) then
   if member.isOneOf(Flags.Method | Flags.Lazy) then
      ...
   else
      ...

 buffer.nonEmpty
```

Reply:
Why did we not have the `!member.isType` condition for methods before? Is that just because `member.isType` is always false with `Flags.Method`?

Reply:
Yes, that's true.